### PR TITLE
New rubocop

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: library/control/src/modules/ProductControl.rb [hunk 1]
```diff
@@ -653,15 +619,11 @@ def RunRequired(stage, mode)
...
-          # proposals
-        elsif !Ops.get_string(one_module, "proposal").nil? &&
-            Ops.get_string(one_module, "proposal", "") != ""
-          next true
-        end
+        name = one_module["name"]
+        proposal = one_module["proposal"]
+
+        next true if name && !name.empty?
+        next true if proposal && !proposal.empty?
```
Comment (dgdavid) [inline library/control/src/modules/ProductControl.rb:626]: Much better now :+1:

File: library/cwm/src/modules/TablePopup.rb [hunk 2]
```diff
@@ -799,6 +791,7 @@ def TableHandle(descr, key, event_descr)
           else
             -1
           end
+          # rubocop:enable Lint/DuplicateBranch
```
Comment (dgdavid) [inline library/cwm/src/modules/TablePopup.rb:794]: Do you think is it a good opportunity for improving it a bit and stop using those legacies `Ops`?

Comment (jreidinger) [reply library/cwm/src/modules/TablePopup.rb:794]: Well, I think TablePopup from CWM is almost not used, so probably not worth it. Also my goal is to not introduce any bugs. So if we improve it, we need also test coverage, which is currently zero.

File: library/cwm/examples/date_time.rb [hunk 3]
```diff
@@ -19,6 +21,8 @@ def label
 
 class EventDate < CWM::DateField
   def initialize
+    super
+
```
Comment (dgdavid) [inline library/cwm/examples/date_time.rb:25]: NP: Just curious about why in yast2-bootloader you kept `textdomain` before `super` and here is the other way around.

File: library/network/src/modules/PortRanges.rb [hunk 4]
```diff
@@ -586,7 +586,7 @@ def FlattenServices(old_list, protocol)
               new_max = this_max_pr
               # the second one partly covers the first one (by its left side)
             elsif Ops.greater_or_equal(min_pr, this_min_pr) &&
-                Ops.less_or_equal(max_pr, this_max_pr)
+                Ops.greater_or_equal(max_pr, this_max_pr)
```
Comment (dgdavid) [inline library/network/src/modules/PortRanges.rb:589]: Nice catch.

File: library/cwm/examples/date_time.rb (hunk 3 above)
Comment (jreidinger) [reply library/cwm/examples/date_time.rb:25]: well, it does not matter. It is order independent as textdomain is added to domain pool and not set exclusively for given object (it will be hard to do it in ruby).

**Review by dgdavid — APPROVED** [review]
LGTM

File: library/network/src/modules/PortRanges.rb (hunk 4 above)
Comment (jreidinger) [reply library/network/src/modules/PortRanges.rb:589]: yeap, rubocop complain that condition is identical and then I spend 5 minutes thinking how it should be correctly.

File: library/cwm/src/modules/TablePopup.rb (hunk 2 above)
Comment (dgdavid) [reply library/cwm/src/modules/TablePopup.rb:794]: Let's keep it as it is. Thanks.
